pkg/dataplane/deployment.go

```diff
@@ -215,34 +216,39 @@ func (d *Deployer) ConditionalDeploy(
 				err.Error()))
 		}
 
-		if ansibleEE.Status.JobStatus == ansibleeev1.JobStatusSucceeded {
+		if ansibleEE.Status.Succeeded != 0 {
 			log.Info(fmt.Sprintf("Condition %s ready", readyCondition))
 			nsConditions.Set(condition.TrueCondition(
 				readyCondition,
 				readyMessage))
 		}
 
-		if ansibleEE.Status.JobStatus == ansibleeev1.JobStatusRunning || ansibleEE.Status.JobStatus == ansibleeev1.JobStatusPending {
-			log.Info(fmt.Sprintf("AnsibleEE job is not yet completed: Execution: %s, Status: %s", ansibleEE.Name, ansibleEE.Status.JobStatus))
+		if ansibleEE.Status.Succeeded == 0 || ansibleEE.Status.Active != 0 && ansibleEE.Status.Failed != 0 {
+			log.Info(fmt.Sprintf("AnsibleEE job is not yet completed: Execution: %s, Active pods: %d", ansibleEE.Name, ansibleEE.Status.Active))
```

Comment:
What about pending job status? I guess we've to check status.active == 0 && status.succeeded == 0 && status.failed == 0?

Reply:
Yeah, true.  I don't think this condition makes sense actually. Because it checks for an Active and a Failed job.

We can probably drop the failed check from here since it's captured in the next section. So this would just become

```suggestion
if ansibleEE.Status.Succeeded > 0 || ansibleEE.Status.Active > 0 {
			log.Info(fmt.Sprintf("AnsibleEE job is not yet completed: Execution: %s, Active pods: %d", ansibleEE.Name, ansibleEE.Status.Active))
}
```

Then the next section will check for if it's failed. In this instance, we just want to log when a job is still active, if it had failed then we would want to log something different.

Reply:
I think there would be scenario when all the three are zero (status.active == 0, status.succeeded == 0 and status.failed == 0) which is when the job is pending and we need to update the status with `.readyWaitingMessage`. I can't see how that is handled here.

Reply:
Right, yes I see what you mean. Like, if the `pod` for the job was in pending. I'll see what the job reports in that case to make sure we have it covered.

Reply:
Looks like even when the pod is pending, the job reports that it's `Active`:

```
❯ oc apply -f test-job.yaml; oc get job pi -o yaml | yq .status.active ; oc get po | grep pi
job.batch/pi created
1
pi-gpwd9                               0/1     Pending      0          0s
```

Reply:
FWIW, I just added a `nodeSelector` that would result in the pod pending to test that:
```
❯ yq .spec.template.spec.nodeSelector test-job.yaml
kubernetes.io/arch: aarch64
```

Only `amd64` nodes in the cluster.

Reply:
Yeah, I don't think it will ever be 0 for pending jobs, the `Active` status field for jobs represents "pending and running pods that aren't terminating":
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/batch/types.go#L524-L528

So this should always be 1 if the Job hasn't culminated in success or failure.